c4gt-demo/backend/common/text_processing.py

```
@@ -0,0 +1,394 @@
+                # Characteristic question
+                template = characteristic_templates[idx % len(characteristic_templates)]
+                prompt = template.format(term=main_keyword)
+                explanation = f"{main_keyword} is characterized by its significance within this educational context."
+            else:
+                # Example question
+                template = example_templates[idx % len(example_templates)]
+                prompt = template.format(term=main_keyword)
+                explanation = f"{main_keyword} represents a key concept illustrated throughout the material."
+            
+            # Shuffle options to avoid position bias
+            all_options = [main_keyword] + distractors[:3]
```

Ruff (0.15.12)

[warning] Consider `[main_keyword, *distractors[:3]]` instead of concatenation

Replace with `[main_keyword, *distractors[:3]]`

(RUF005)